samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -84,11 +112,195 @@ const ShareView = () => {
         }
     }
 
+    // Initiate facebook login.
+    const facebookLogin = () => {
+        fbAuthentication() // This method get a client-side token for Facebook
+            .then((result) => {
+                return getServerSideTokenFb(result.idToken); // This method get a face book user profile details.
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    // Get client side token for facebook.
+    const fbAuthentication = () => {
+        var facebookAppId = process.env.REACT_APP_FACEBOOK_APP_ID;
+        let redirectUri = window.location.origin + "/facebook-auth-end";
+
+        return new Promise((resolve, reject) => {
+            microsoftTeams.authentication.authenticate({
+                url: `https://www.facebook.com/v12.0/dialog/oauth?client_id=${facebookAppId}&redirect_uri=${redirectUri}&state=1234535`,
```

Comment:
State shouldn't be a static value, as it could lead to Cross-site Request Forgery. We should instead set a random value here, and verify that it matches the state returned in the auth-end call.

Reply:
Updated your feedback.